File: 2573-remove-nodes-from-linked-list/remove-nodes-from-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev=None
        curr=head
        nextn=None
        while (curr):
            nextn=curr.next
            curr.next=prev
            prev=curr
            curr=nextn
        head=prev

        newHead=None
        p=head
        max_=float('-inf')
        while (p):
            if p.val>=max_:
                temp=ListNode(p.val)
                temp.next=newHead
                newHead=temp
                max_=p.val
            p=p.next
        return newHead



        

File: 2573-remove-nodes-from-linked-list/remove-nodes-from-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNodes(struct ListNode* head) {
    struct ListNode* curr = head;
    struct ListNode* prev = NULL;
    struct ListNode* nextn = NULL;
    struct ListNode* newHead = NULL;
    int max;
    
    // Inversing the list
    while (curr) {
        nextn = curr->next;
        curr->next = prev;
        prev = curr;
        curr = nextn;
    }
    head = prev;

    // Iterating thgouth the reversed list saving the max nodes
    newHead = malloc(sizeof(struct ListNode));
    newHead->val = head->val;
    newHead->next=NULL;

    curr = head->next;
    max = newHead->val;
    while (curr) {
        if (curr->val >= max) {
            nextn = malloc(sizeof(struct ListNode));
            nextn->val = curr->val;
            nextn->next = newHead;
            newHead = nextn;
            max=curr->val;
        }
        curr = curr->next;
    }

    return newHead;
}